dex-it/src/test/scala/com/wavesplatform/it/matcher/api/http/cancel/CancelOrdersByAddressAndIdsSpec.scala

```diff
@@ -71,16 +108,43 @@ class CancelOrdersByAddressAndIdsSpec extends MatcherSuiteBase with ApiKeyHeader
 
     "should return an error when address is not a correct base58 string" in {
       validateMatcherError(
-        dex1.rawApi.cancelOrdersByIdsWithKey("null", placeAndGetIds(3)),
+        dex1.rawApi.cancelOrdersByIdsWithKeyOrSignature("null", placeAndGetIds(3)),
         StatusCode.BadRequest,
         InvalidAddress.code,
         "Provided address is not correct, reason: Unable to decode base58: requirement failed: Wrong char 'l' in Base58 string 'null'"
       )
     }
 
-    shouldReturnErrorWithoutApiKeyHeader(dex1.rawApi.cancelOneOrderWithKey(order.idStr(), headers = Map.empty))
+    "should return order not found if address path parameter and publicKey header are from different accounts" in {
+      val o = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+      placeAndAwaitAtDex(o)
+      val ts = System.currentTimeMillis
+      val sign = crypto.sign(alice, alice.publicKey ++ Longs.toByteArray(ts))
+
+      val r =
+        validate200Json(dex1.rawApi.cancelOrdersByIdsWithKeyOrSignature(bob.stringRepr, Set(o.idStr()), mkHeaders(alice.publicKey, ts, sign)))
+
+      r.message.head should have size 1
+      r.message.head.foreach {
+        case util.Left(HttpError(e, m, _, _, _, _)) =>
+          e shouldBe OrderNotFound.code
```

Comment:
probably we shouldn't go to address actor for this and just return invalidUserPublicKey ad we do here:

```
userPublicKey match {
              case Some(upk) if upk.toAddress != address => invalidUserPublicKey
```
@vsuharnikov what do you think?

Reply:
This would be hard to implement because of how `signedGet` works. Let's stick the current code.